Overview
========

Inputs
------

+-------------------+------------+----------------------------------------------------------------------+-----------------------+
| Name              | Type       | Description                                                          | Example               |  
+===================+============+======================================================================+=======================+
| FastQ files       | Input file | Gzipped FastQ files generated by human transcriptome sequencing      | *.fastq.gz or *.fq.gz |
+-------------------+------------+----------------------------------------------------------------------+-----------------------+

Outputs
-------

The Rapid RNA-Seq pipeline produces the following outputs:

+------------------------------------+--------------------------------------------------+
| Name                               | Description                                      |
+====================================+==================================================+
| Peak region file (.bed)            | File containing putative gene fusions.           | 
+------------------------------------+--------------------------------------------------+
| Compressed .bed file (.bb)         | File containing putative gene fusions.           | 
+------------------------------------+--------------------------------------------------+
| Coverage file (.bw)                | bigWig file containing coverage information.     |
+------------------------------------+--------------------------------------------------+
| Splice junction read counts (.txt) | Read counts for the splice junction detected.    |
+------------------------------------+--------------------------------------------------+
| Interactive fusion visualization   | Fusion visualization produced by ProteinPaint.   |
+------------------------------------+--------------------------------------------------+
| Interactive coverage visualization | Coverage visualization produced by ProteinPaint. |
+------------------------------------+--------------------------------------------------+

Process
-------

Synopsis
++++++++

1. FastQC (    ) is used with standard parameters to check the sequencing quality.
2. The raw sequence data is aligned to the selected reference genome using BWA (    )alignment.
3. Duplicate or multiple mapped reads are removed.
4. Quality check is done for enrichment and cross correlation plot is created.
   * Broad peaks are called using **SICER** 
   * Narrow peaks are called using **MACS2**
5. Peaks are filtered using the blacklist and the clean.bed file is created.
5. Summary statistics are run and visualization files are created, as well as a shortcut to ProteinPaint. 
